src/segments/umbraco.go

```diff
@@ -0,0 +1,231 @@
+package segments
+
+import (
+	"encoding/xml"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Umbraco struct {
+	props properties.Properties
+	env   platform.Environment
+
+	FoundUmbraco    bool
+	IsModernUmbraco bool
+	IsLegacyUmbraco bool
+	Version         string
+}
+
+const (
+	umbracoFolderName = "umbraco"
+	umbracoWebConfig  = "web.config"
+)
+
+type CSProj struct {
+	PackageReferences []struct {
+		Name    string `xml:"Include,attr"`
+		Version string `xml:"Version,attr"`
+	} `xml:"ItemGroup>PackageReference"`
+}
+
+type WebConfig struct {
+	AppSettings []struct {
+		Key   string `xml:"key,attr"` // TODO: What happens if the web.config has the attribute as uppercase Key="" ?
+		Value string `xml:"value,attr"`
+	} `xml:"appSettings>add"`
+}
+
+type FindUmbracoResult struct {
+	FoundUmbracoFolder bool
+	FoundWebConfig     bool
+	FoundCSProj        bool
+	FilePath           string
+}
+
+func (u *Umbraco) Enabled() bool {
+	u.env.Debug("UMBRACO: Checking if we enable segment")
+
+	// Checks if the current folder contains an umbraco folder
+	// If not it then checks each parent until the root
+	findUmbracoResults, err := u.TryFindUmbracoInParentDirsOrSelf()
+
+	if err != nil {
+		u.env.Debug("UMBRACO: Error while searching for Umbraco folder and files")
+		u.env.Debug(err.Error())
+		return false
+	}
+
+	if !findUmbracoResults.FoundUmbracoFolder {
```

Comment:
I would remove the `FoundUmbracoFolder` and return an error in case there's no match. If you only log relevant errors in `TryFindUmbracoInParentDirsOrSelf()` and use the internal error for deciding to continue, it makes the `TryFindUmbracoInParentDirsOrSelf()` function simpler and you can also omit the struct return and instead just return 3 variables instead. You don't need the struct overhead.

Reply:
Removed the struct and just return the 3 variables and make an error if nothing is found to remove the FoundUmbracoFolder check